web/html/src/manager/admin/config/use-monitoring-api.ts

```diff
@@ -0,0 +1,127 @@
+import { useState } from "react";
+import Network, { JsonResult } from "utils/network";
+import { MessageType } from "components/messages";
+
+type ExportersResultType = {
+  exporters: {
+    [key: string]: boolean;
+  };
+  messages: {
+    [key: string]: string;
+  };
+};
+
+function isRestartNeeded(data: ExportersResultType) {
+  return Object.keys(data.exporters).some(key => data.messages[key] === "restart");
+}
+
+const useMonitoringApi = () => {
+  const [action, setAction] = useState<null | "checking" | "enabling" | "disabling">(null);
+  const [exportersStatus, setExportersStatus] = useState<
+    | {
```

Comment:
I think there's an extra `|` at the beginning. Is it still valid?

Reply:
The extra pipe is valid syntax (although imo not very pretty) that can be used for multiline type expressions. It's formatted this way by Prettier here, not having the extra `|` would be valid as well.  
There's an example of this from [the official docs](https://www.typescriptlang.org/docs/handbook/unions-and-intersections.html#discriminating-unions):  

```ts
type NetworkState =
  | NetworkLoadingState
  | NetworkFailedState
  | NetworkSuccessState;
```

Reply:
Yeah, later on, I saw some other examples in our code as well. No big deal then, I was just curious.